### PR TITLE
Upcoming events

### PR DESCRIPTION
**Pull request by oliveiraigorm**

### CODE REVIEW

**Review by frozenbanana — CHANGES_REQUESTED** [review]
Can we get the coming three events instead of within a timespan?
I prefer a function that formats with template strings.

Also, the events are bound to the calendar name "Personal" we should consider the case when the user has more than one calendar.


```js
const getEventsUrl = (userName, calendarName) => (`/remote.php/dav/calendars/${userName}/${calendarName}?accept=jcal&componentType=VEVENT`)
```

then

```js
mounted() {
axios
  .get(getEventsUrl(userName, calName))
  ..
```


Also, I think it makes sense to have util functions to fetch the data we want from the repsonse. It makes it more readable. 

`id: element[1][4][3],` -> `id: getId(element)`

`title: element[1][8][3],` -> `id: getTitle(element)`

etc.


Also, I prefer to keep functions pure so instead of mutating `calendarEvents` with unshift, use `.map(fn)`

```js
calendarEvents = getEventsObjects(response.data).map(element=> ({id: getId(element), title: getTitle(element), ... }) )
```

**Comment by oliveiraigorm** [comment]
> Can we get the coming three events instead of within a timespan?

No the plugin can only limit by time span. What I could do is to get the events within a certain time span or only the future events, to avoid getting the entire calendar, and limit to display only the 3 more recent ones.

File: src/components/Events.vue [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			<li>{{ event.title }} {{ event.tstart }}</li>
+			<h3>{{ event.description }}</h3>
+			<h4>{{ event.tstart }} - {{ event.tend }}</h4>
+		</ul>
+	</div>
+</template>
+
+<script>
+import axios from "@nextcloud/axios";
+
```
Comment (frozenbanana) [inline src/components/Events.vue:13]: import {getId, getUserName, .. } from '../utils';

File: src/utils.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+export const userName = document
+	.querySelector("head")
+	.getAttribute("data-user-displayname");
```
Comment (frozenbanana) [inline src/utils.js:3]: `.getAttribute("data-user-displayname")` should be `.getAttribute("data-user");`

**Review by frozenbanana — APPROVED** [review]

**Pull request merged**